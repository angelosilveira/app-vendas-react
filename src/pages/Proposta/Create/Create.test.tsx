import { screen, waitFor } from '@testing-library/react';
import { renderWithTheme } from 'src/utils/tests/renderWithTheme';
import userEvent from '@testing-library/user-event';
import { CreateProposal } from '.';

describe('<CreateProposal />', () => {
  it('should render page create proposal', () => {
    renderWithTheme(<CreateProposal />);

    expect(screen.getByText('Proposta de Venda')).toBeInTheDocument();
  });

  it('should error fields empty when click on button', async () => {
    const { container } = renderWithTheme(<CreateProposal />);

    const buttonSave = screen.getByText('Salvar');

    userEvent.click(buttonSave);

    await waitFor(() => {
      expect(
        screen.getByText('Campo Data da proposta é obrigatório')
      ).toBeInTheDocument();
      expect(
        screen.getByText('Campo CNPJ/CPF é obrigatório')
      ).toBeInTheDocument();
      expect(
        screen.getByText('Campo Nome do Títular é obrigatório')
      ).toBeInTheDocument();
      expect(
        screen.getByText('Campo Estado Cívil é obrigatório')
      ).toBeInTheDocument();
      expect(
        screen.getByText('Campo Estado é obrigatório')
      ).toBeInTheDocument();
      expect(
        screen.getByText('Campo Cidade é obrigatório')
      ).toBeInTheDocument();
      expect(
        screen.getByText('Campo Bairro é obrigatório')
      ).toBeInTheDocument();
      expect(
        screen.getByText('Campo Endereço é obrigatório')
      ).toBeInTheDocument();
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
