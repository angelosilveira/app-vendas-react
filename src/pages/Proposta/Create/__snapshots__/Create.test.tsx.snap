// Vitest Snapshot v1

exports[`<CreateProposal /> > should error fields empty when click on button 1`] = `
.c0 {
  padding-left: 15px;
  padding-right: 15px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
}

.c1 {
  border-bottom: 1px solid #e9e9e9;
  margin-bottom: 1rem;
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.c1 h2 {
  font-size: 2.0rem;
  color: #030517;
  margin-bottom: 0;
}

.c6 {
  margin-bottom: 1rem;
}

.c6 input {
  display: block;
  width: 100%;
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #212529;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  -webkit-appearance: none;
  -moz-appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border-radius: 0.25rem;
  -webkit-transition: border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  height: 38px;
}

.c6 input:focus {
  color: #212529;
  background-color: #fff;
  border-color: #86b7fe;
  outline: 0;
}

.c7 {
  color: #f00;
  font-weight: bold;
  margin-top: 0.25rem;
  font-size: 1.2rem;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  font-weight: 400;
  line-height: 1.5;
  color: #212529;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: transparent;
  border: 1px solid transparent;
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  border-radius: 0.25rem;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  font-size: 1.2rem;
  color: #fff;
  background-color: #0d6efd;
  border-color: #0d6efd;
}

.c10 svg {
  font-size: 1.6rem;
  margin-right: 5px;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  font-weight: 400;
  line-height: 1.5;
  color: #212529;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: transparent;
  border: 1px solid transparent;
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  border-radius: 0.25rem;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  font-size: 1.2rem;
  color: #000;
  background-color: #ffc107;
  border-color: #ffc107;
}

.c11 svg {
  font-size: 1.6rem;
  margin-right: 5px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: calc(100vh - 150px);
}

.c2 .input-date {
  margin-bottom: 1rem;
}

.c2 .input-date label {
  display: inline-block;
}

.c2 .input-date input {
  display: block;
  height: 38px;
  width: 100%;
  padding: 1rem 0.75rem;
}

.c2 footer button + button {
  margin-top: 10px;
}

.c5 {
  color: #f00;
  font-weight: bold;
  margin-top: 0.25rem;
  font-size: 1.2rem;
}

.c3 {
  position: relative;
  height: calc(100vh - 215px);
  margin-right: -15px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c3 .cover-bar {
  position: absolute;
  background: #f4f4f4;
  height: 100%;
  top: 0;
  right: 0;
  width: 0.4em;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
  opacity: 1;
  pointer-events: none;
}

.c3:hover .cover-bar {
  opacity: 0;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
}

.c4 {
  position: absolute;
  width: 100%;
  left: 0;
  overflow-y: scroll;
  height: 100%;
  padding-right: 10px;
}

.c4::-webkit-scrollbar {
  width: 0.4em;
}

.c4::-webkit-scrollbar,
.c4::-webkit-scrollbar-thumb {
  overflow: visible;
  border-radius: 4px;
}

.c4::-webkit-scrollbar-thumb {
  background: rgba(0,0,0,0.2);
}

.c8 {
  margin-bottom: 1rem;
}

.c8 select {
  display: block;
  width: 100%;
  padding: 1rem 0.5rem;
  font-size: 1.2rem;
  font-weight: 600;
  line-height: 1.5;
  color: #212529;
  background-color: #fff;
  background-repeat: no-repeat;
  background-position: right 0.75rem center;
  background-size: 16px 12px;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  -webkit-transition: border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
}

.c9 {
  color: #f00;
  font-weight: bold;
  margin-top: 0.25rem;
  font-size: 1.2rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h2>
      Proposta de Venda
    </h2>
  </div>
  <form
    class="c2"
  >
    <section>
      <div
        class="c3"
        data-testid="scroll"
      >
        <div
          class="c4"
        >
          <div
            class="input-date"
          >
            <label>
              Data da proposta*
            </label>
            <input
              name="date_proposal"
              type="date"
            />
            <div
              class="c5"
            >
              Campo Data da proposta é obrigatório
            </div>
          </div>
          <div
            class="c6"
          >
            <label
              for="cnpj_cpf"
            >
              CNPJ/CPF*
            </label>
            <input
              aria-label="cnpj_cpf"
              id="cnpj_cpf"
              placeholder="CNPJ/CPF*"
            />
            <div
              class="c7"
            >
              Campo CNPJ/CPF é obrigatório
            </div>
          </div>
          <div
            class="c6"
          >
            <label
              for="name"
            >
              Nome do Titular*
            </label>
            <input
              aria-label="name"
              id="name"
              placeholder="Nome do Titular*"
            />
            <div
              class="c7"
            >
              Campo Nome do Títular é obrigatório
            </div>
          </div>
          <div
            class="c6"
          >
            <label
              for="rg"
            >
              IE/RG
            </label>
            <input
              aria-label="rg"
              id="rg"
              placeholder="IE/RG"
            />
          </div>
          <div
            class="c8"
          >
            <label>
              Estado Cívil*
            </label>
            <select
              data-testid="marital"
              name="marital"
            >
              <option
                disabled=""
                value=""
              >
                Selecione o estado cívil
              </option>
              <option
                value="solteiro"
              >
                Solteiro(a)
              </option>
              <option
                value="casado"
              >
                Casado(a)
              </option>
              <option
                value="divorciado"
              >
                Divorciado(a)
              </option>
              <option
                value="viuvo"
              >
                Viúvo(a)
              </option>
            </select>
            <div
              class="c9"
            >
              Campo Estado Cívil é obrigatório
            </div>
          </div>
          <div
            class="c6"
          >
            <label
              for="phone"
            >
              Telefone
            </label>
            <input
              aria-label="phone"
              id="phone"
              placeholder="Telefone"
            />
          </div>
          <div
            class="input-date"
          >
            <label>
              Data de nascimento
            </label>
            <input
              name="date_birth"
              type="date"
            />
          </div>
          <div
            class="c6"
          >
            <label
              for="cel"
            >
              Celular
            </label>
            <input
              aria-label="cel"
              id="cel"
              placeholder="Celular"
            />
          </div>
          <div
            class="c8"
          >
            <label>
              Sexo
            </label>
            <select
              name="sex"
            >
              <option
                disabled=""
                value=""
              >
                Selecione o sexo
              </option>
              <option
                value="masculino"
              >
                Masculino
              </option>
              <option
                value="feminino"
              >
                Feminino
              </option>
              <option
                value="outro"
              >
                Outro
              </option>
            </select>
          </div>
          <div
            class="c6"
          >
            <label
              for="email"
            >
              E-mail
            </label>
            <input
              aria-label="email"
              id="email"
              placeholder="E-mail"
              type="email"
            />
          </div>
          <div
            class="c8"
          >
            <label>
              Estado*
            </label>
            <select
              name="state"
            >
              <option
                disabled=""
                value=""
              >
                Selecione um estado
              </option>
              <option
                value="AC"
              >
                Acre
              </option>
              <option
                value="AL"
              >
                Alagoas
              </option>
              <option
                value="AP"
              >
                Amapá
              </option>
              <option
                value="AM"
              >
                Amazonas
              </option>
              <option
                value="BA"
              >
                Bahia
              </option>
              <option
                value="CE"
              >
                Ceará
              </option>
              <option
                value="DF"
              >
                Distrito Federal
              </option>
              <option
                value="ES"
              >
                Espírito Santo
              </option>
              <option
                value="GO"
              >
                Goiás
              </option>
              <option
                value="MA"
              >
                Maranhão
              </option>
              <option
                value="MT"
              >
                Mato Grosso
              </option>
              <option
                value="MS"
              >
                Mato Grosso do Sul
              </option>
              <option
                value="MG"
              >
                Minas Gerais
              </option>
              <option
                value="PA"
              >
                Pará
              </option>
              <option
                value="PB"
              >
                Paraíba
              </option>
              <option
                value="PR"
              >
                Paraná
              </option>
              <option
                value="PE"
              >
                Pernambuco
              </option>
              <option
                value="PI"
              >
                Piauí
              </option>
              <option
                value="RJ"
              >
                Rio de Janeiro
              </option>
              <option
                value="RN"
              >
                Rio Grande do Norte
              </option>
              <option
                value="RS"
              >
                Rio Grande do Sul
              </option>
              <option
                value="RO"
              >
                Rondônia
              </option>
              <option
                value="RR"
              >
                Roraima
              </option>
              <option
                value="SC"
              >
                Santa Catarina
              </option>
              <option
                value="SP"
              >
                São Paulo
              </option>
              <option
                value="SE"
              >
                Sergipe
              </option>
              <option
                value="TO"
              >
                Tocantins
              </option>
            </select>
            <div
              class="c9"
            >
              Campo Estado é obrigatório
            </div>
          </div>
          <div
            class="c8"
          >
            <label>
              Cidade*
            </label>
            <select
              name="city"
            >
              <option
                disabled=""
                value=""
              >
                Selecione uma cidade
              </option>
            </select>
            <div
              class="c9"
            >
              Campo Cidade é obrigatório
            </div>
          </div>
          <div
            class="c6"
          >
            <label
              for="neighborhood"
            >
              Bairro*
            </label>
            <input
              aria-label="neighborhood"
              id="neighborhood"
              placeholder="Bairro*"
            />
            <div
              class="c7"
            >
              Campo Bairro é obrigatório
            </div>
          </div>
          <div
            class="c6"
          >
            <label
              for="address"
            >
              Endereço*
            </label>
            <input
              aria-label="address"
              id="address"
              placeholder="Endereço*"
            />
            <div
              class="c7"
            >
              Campo Endereço é obrigatório
            </div>
          </div>
          <div
            class="c6"
          >
            <label
              for="number"
            >
              Número
            </label>
            <input
              aria-label="number"
              id="number"
              placeholder="Número"
            />
          </div>
          <div
            class="c6"
          >
            <label
              for="zipcode"
            >
              CEP
            </label>
            <input
              aria-label="zipcode"
              id="zipcode"
              placeholder="CEP"
            />
          </div>
          <div
            class="c6"
          >
            <label
              for="complement"
            >
              Complemento
            </label>
            <input
              aria-label="complement"
              id="complement"
              placeholder="Complemento"
            />
          </div>
        </div>
        <div
          class="cover-bar"
        />
      </div>
    </section>
    <footer>
      <button
        class="c10"
        type="submit"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 21h14a2 2 0 0 0 2-2V8a1 1 0 0 0-.29-.71l-4-4A1 1 0 0 0 16 3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2zm10-2H9v-5h6zM13 7h-2V5h2zM5 5h2v4h8V5h.59L19 8.41V19h-2v-5a2 2 0 0 0-2-2H9a2 2 0 0 0-2 2v5H5z"
          />
        </svg>
        Salvar
      </button>
      <button
        class="c11"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 10h6c1.654 0 3 1.346 3 3s-1.346 3-3 3h-3v2h3c2.757 0 5-2.243 5-5s-2.243-5-5-5H9V5L4 9l5 4v-3z"
          />
        </svg>
        Voltar
      </button>
    </footer>
  </form>
</div>
`;
